test/e2e/certmanager/certmanager_test.go

```diff
@@ -0,0 +1,134 @@
+		ns           *corev1.Namespace
+		defaultRf    *kueue.ResourceFlavor
+		localQueue   *kueue.LocalQueue
+		clusterQueue *kueue.ClusterQueue
+	)
+
+	ginkgo.BeforeAll(func() {
+		configurationUpdate := time.Now()
+		config := defaultKueueCfg.DeepCopy()
+		config.InternalCertManagement = &v1beta1.InternalCertManagement{
+			Enable: ptr.To(false),
+		}
+		util.ApplyKueueConfiguration(ctx, k8sClient, config)
+		util.RestartKueueController(ctx, k8sClient)
+		util.WaitForKueueAvailability(ctx, k8sClient)
+		ginkgo.GinkgoLogr.Info("Kueue configuration updated", "took", time.Since(configurationUpdate))
+	})
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{GenerateName: "e2e-cert-manager-"},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+
+		defaultRf = testing.MakeResourceFlavor("default").Obj()
+		gomega.Expect(k8sClient.Create(ctx, defaultRf)).Should(gomega.Succeed())
+
+		clusterQueue = testing.MakeClusterQueue("cluster-queue").
+			ResourceGroup(*testing.MakeFlavorQuotas(defaultRf.Name).
+				Resource(corev1.ResourceCPU, "2").
+				Resource(corev1.ResourceMemory, "2G").Obj()).Obj()
+		gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+
+		localQueue = testing.MakeLocalQueue("main", ns.Name).ClusterQueue("cluster-queue").Obj()
+		gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, clusterQueue, true, util.LongTimeout)
+		util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, defaultRf, true, util.LongTimeout)
+	})
+
+	ginkgo.When("CertManager is Enabled", func() {
+		ginkgo.It("should admit a Job", func() {
+			testJob := testingjob.MakeJob("test-job", ns.Name).
+				Queue("main").
+				Suspend(false).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, testJob)).To(gomega.Succeed())
+
+			verifyAdmission(
+				testJob.Name,
+				ns.Name,
+				workloadjob.GetWorkloadNameForJob,
+				schema.GroupVersionKind{Group: "batch", Version: "v1", Kind: "Job"},
+			)
+		})
+	})
+})
+
+func verifyAdmission(name, namespace string, workloadNameFn func(string, types.UID) string, gvk schema.GroupVersionKind) {
+	ginkgo.By("Checking resource status", func() {
+		key := types.NamespacedName{Name: name, Namespace: namespace}
+		gomega.Eventually(func(g gomega.Gomega) {
+			obj := waitForObject(ctx, key, gvk)
+			suspend, _, _ := unstructured.NestedBool(obj.Object, "spec", "suspend")
+			g.Expect(suspend).To(gomega.BeFalse())
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
+	})
+
+	ginkgo.By("Verifying workload admission", func() {
+		key := types.NamespacedName{Name: name, Namespace: namespace}
+		obj := waitForObject(ctx, key, gvk)
+		wlLookupKey := types.NamespacedName{}
+		wlLookupKey = types.NamespacedName{Name: workloadNameFn(name, obj.GetUID()), Namespace: namespace}
```

Comment:
use `workloadjob.GetWorkloadNameForJob(` instead. I see the intention of making this generic, but this is not that idiomatic, and makes the code inconsistent. I'm open to have a PR which introduces some generic helper functions, but then I would use them throughout the codebase.

Reply:
Sure, sounds good!